### PR TITLE
XIVY-14725 Add AttributeBrowser to find attributes of type

### PR DESCRIPTION
**Pull request by ivy-edp**
I have now created a browser for the dynamic label and value, with which you can find the attributes from the type of the List of Objects selected above. If the field is left empty, the entire object is used (we check in the backend, if empty, the name inside var which is ‘variable’ is used).

selectComponentFInal

### CODE REVIEW

**Comment by ivy-edp** [comment]
What do you think of the naming. Is it okay or can I change something to make it more low-code friendly?

File: packages/editor/src/components/blocks/select/Select.tsx [hunk 1]
```diff
@@ -40,14 +40,14 @@ export const SelectComponent: ComponentConfig<SelectProps> = {
     dynamicItemsLabel: {
       subsection: 'Dynamic Options',
       label: 'Dynamic Label',
```
Comment (ivy-lli) [inline packages/editor/src/components/blocks/select/Select.tsx:42]: maybe object label is better

File: packages/editor/src/components/blocks/select/Select.tsx [hunk 2]
```diff
@@ -40,14 +40,14 @@ export const SelectComponent: ComponentConfig<SelectProps> = {
...
-      type: 'text',
-      options: { placeholder: 'Enter fitting attribute' }
+      type: 'textBrowser',
+      options: { onlyAttributesVariable: true, placeholder: 'Enter attribute (or leave blank to select entire object)' }
```
Comment (ivy-lli) [inline packages/editor/src/components/blocks/select/Select.tsx:44]: maybe you can also think about to implement the `hide` function. With it you can control if the option should be visible. E.g. in this case the label and value only makes sense if you have defined the options value.

File: packages/editor/src/data/variable-tree-data.test.ts [hunk 3]
```diff
@@ -128,13 +128,25 @@ describe('variableTreeData', () => {
...
+test('findListVariable of endless', () => {
   const list = findListVariables(endlessParamInfo);
   expect(list.length).toEqual(1);
   expect(list[0].value).toEqual('param.Endless.endlessList');
   expect(list[0].info).toEqual('List<demo.Endless>');
 });
 
+test('findRelevantAttributesOfType of List<demo.Endless>', () => {
+  const list = findRelevantAttributesOfType(endlessParamInfo, 'param.Endless.endlessList');
+  console.log(list);
```
Comment (ivy-lli) [inline packages/editor/src/data/variable-tree-data.test.ts:140]: remove

File: packages/editor/src/data/variable-tree-data.ts [hunk 4]
```diff
@@ -105,3 +111,68 @@ export const findListVariables = (variableInfo: VariableInfo, maxDepth: number =
 
   return result;
 };
+
+export function findRelevantAttributesOfType(
```
Comment (ivy-lli) [inline packages/editor/src/data/variable-tree-data.ts:115]: why not simple `findAttributesOfType`?

File: packages/editor/src/editor/properties/fields/InputFieldWithBrowser.tsx [hunk 5]
```diff
@@ -13,7 +13,8 @@ export const InputFieldWithBrowser = ({
   options
 }: InputFieldProps & { options?: TextBrowserFieldOptions }) => {
   const [open, setOpen] = useState(false);
-  const attrBrowser = useAttributeBrowser(!!options?.displayOnlyListTypes);
+  const attrBrowser = useAttributeBrowser(!!options?.displayOnlyListTypes, !!options?.onlyAttributesVariable);
```
Comment (ivy-lli) [inline packages/editor/src/editor/properties/fields/InputFieldWithBrowser.tsx:16]: streamline options: 
- onlyListTypes
- onlyAttributes

**Review by ivy-lli — APPROVED** [review]

File: packages/editor/src/components/blocks/select/Select.tsx (hunk 1 above)
Comment (ivy-edp) [reply packages/editor/src/components/blocks/select/Select.tsx:42]: like this?
grafik

File: packages/editor/src/components/blocks/select/Select.tsx (hunk 2 above)
Comment (ivy-edp) [reply packages/editor/src/components/blocks/select/Select.tsx:44]: Yes I agree. I will try to implement this in the next PR 👍

File: packages/editor/src/data/variable-tree-data.ts (hunk 4 above)
Comment (ivy-edp) [reply packages/editor/src/data/variable-tree-data.ts:115]: 👍

File: packages/editor/src/components/blocks/select/Select.tsx (hunk 1 above)
Comment (ivy-lli) [reply packages/editor/src/components/blocks/select/Select.tsx:42]: and the value too

**Pull request merged**